metaflow/plugins/argo/argo_client.py

```diff
@@ -203,3 +203,146 @@ def schedule_workflow_template(self, name, schedule=None):
+                    raise ArgoClientException(
+                        json.loads(e.body)["message"]
+                        if e.body is not None
+                        else e.reason
+                    )
+            else:
+                raise ArgoClientException(
+                    json.loads(e.body)["message"] if e.body is not None else e.reason
+                )
+        try:
+            return self._client.CustomObjectsApi().replace_namespaced_custom_object(
+                group=self._group,
+                version=self._version,
+                namespace=self._namespace,
+                plural="sensors",
+                name=name,
+                body=sensor_template,
+            )
+        except self._client.rest.ApiException as e:
+            raise ArgoClientException(
+                json.loads(e.body)["message"] if e.body is not None else e.reason
+            )
+
+    def remove_existing_trigger_on_template(self, name, dry_run=False):
+        if dry_run:
+            return self._describe_remove_existing_trigger_on_template(name)
+        else:
+            return self._remove_existing_trigger_on_template(name)
+
+    def _remove_existing_trigger_on_template(self, name):
+        try:
+            self._client.CustomObjectsApi().delete_namespaced_custom_object(
+                group=self._group,
+                version=self._version,
+                namespace=self._namespace,
+                plural="sensors",
+                name=name,
+                grace_period_seconds=0,
+            )
+            return (True,)
+        except self._client.rest.ApiException as e:
+            if e.status == 404:
+                return (False,)
+
+            raise ArgoClientException(
+                json.loads(e.body)["message"] if e.body is not None else e.reason
+            )
+
+    def _describe_remove_existing_trigger_on_template(self, name):
+        try:
+            results = self._client.CustomObjectsApi().list_namespaced_custom_object(
+                group=self._group,
+                version=self._version,
+                namespace=self._namespace,
+                plural="sensors",
+                watch=False,
+            )
+            templates = results["items"]
+            for t in templates:
+                if t["metadata"]["name"] == name:
+                    annotations = t["metadata"]["annotations"]
+                    triggered_by = annotations["metaflow/triggered_by"]
+                    trigger_status = annotations["metaflow/trigger_status"]
+                    return (True, triggered_by, trigger_status)
+            return (False,)
+        except self._client.rest.ApiException as e:
+            if e.status == 404:
+                return (False,)
+
+            raise ArgoClientException(
+                json.loads(e.body)["message"] if e.body is not None else e.reason
+            )
+
+    def remove_existing_workflow_template(self, flow_name, dry_run=False):
```

Comment:
If we drop `explicit deletes` for now, we should be able to drop these methods as well